app/views/versions/index.html.erb

```diff
@@ -1,12 +1,22 @@
-<h1>Versions for <%= params.require(:item_type) %> : <%= params.require(:item_id) %></h1>
+<% title = "Versions for #{params.require(:item_type)} #{params.require(:item_id)}" %>
+<%= content_for :page_title, title %>
+<h1><%= title %></h1>
 
 <table class="table">
   <tr>
     <th>Date</th>
     <th>Modified by</th>
-    <th>Data</th>
+    <th>Previous version</th>
   </tr>
-  <% @versions.each do |v| %>
+  <%
+    empty = {}.to_yaml
+    current = YAML.load(@versions[0]&.object || empty)
+    @versions.each_with_index do |v, i|
+      previous = YAML.load(@versions[i + 1]&.object || empty)
+      different = (previous.keys + current.keys).uniq.map { |k| [k, previous[k], current[k]] }.select { |k, p, c| p != c }
+      current = previous
+    %>
```

Comment:
How about we put this in the controller or a helper?  Seems like a little too much logic to have in a view.